apiserver/pkg/http/client.go

```diff
@@ -638,29 +648,78 @@ func (krc *KuberayAPIServerClient) doDelete(deleteURL string) (*rpcStatus.Status
 }
 
 func (krc *KuberayAPIServerClient) executeRequest(httpRequest *http.Request, URL string) ([]byte, *rpcStatus.Status, error) {
-	response, err := krc.httpClient.Do(httpRequest)
-	if err != nil {
-		return nil, nil, fmt.Errorf("failed to execute http request for url '%s': %w", URL, err)
-	}
-	defer func() {
-		if closeErr := response.Body.Close(); closeErr != nil {
-			klog.Errorf("Failed to close http response body because %+v", closeErr)
+	// Record the last error and status got
+	var lastErr error
+	var lastStatus *rpcStatus.Status
+
+	for attempt := 0; attempt < krc.maxRetry; attempt++ {
+		response, err := krc.httpClient.Do(httpRequest)
+		// do not retry for Go Error
+		if err != nil {
+			lastStatus = nil
+			lastErr = fmt.Errorf("failed to execute http request for url '%s': %w", URL, err)
+			break
 		}
-	}()
-	bodyBytes, err := io.ReadAll(response.Body)
-	if err != nil {
-		return nil, nil, fmt.Errorf("failed to read response body bytes: %w", err)
-	}
-	if response.StatusCode != http.StatusOK {
-		status, err := krc.extractStatus(bodyBytes)
+
+		defer func() {
+			if closeErr := response.Body.Close(); closeErr != nil {
+				klog.Errorf("Failed to close http response body because %+v", closeErr)
```

Comment:
nit: I understand it's not your issue, but let's keep error message consistent and use either `Failed` or `failed`.